ipme.c

```diff
@@ -52,20 +52,35 @@ int ipme_init()
   byte_copy(&ix.ip,4,"\0\0\0\0");
   if (!ipalloc_append(&ipme,&ix)) { return 0; }
   if ((s = socket(AF_INET,SOCK_STREAM,0)) == -1) return -1;
- 
-  len = 256;
-  for (;;) {
-    if (!stralloc_ready(&buf,len)) { close(s); return 0; }
-    buf.len = 0;
+
+  ifc.ifc_buf = 0;
+  ifc.ifc_len = 0;
+
+  /* first pass: just ask what the correct length for all addresses is */
+  len = 0;
+  if (ioctl(s,SIOCGIFCONF,&ifc) >= 0 && ifc.ifc_len > 0) { /* > is for System V */
+    if (!stralloc_ready(&buf,ifc.ifc_len)) { close(s); return 0; }
```

Comment:
See [man netdevice](https://linux.die.net/man/7/netdevice). This is a union, the actual thing is ```struct ifreq*```, but one can assign to char*, probably for correct sizing (i.e. no factor needed to get from the amount of memory in buf to what is allocated).

Reply:
That's a helpful reference, thanks! My main question here is really only about null-termination, because I noticed myself assuming that if we have an `ifc_len`, our `ifc_buf` won't need or have a trailing `\0`. Just wanted to check this assumption.